services/comm.cpp

```diff
@@ -1859,13 +1859,14 @@ void JobDoneMsg::send_to_channel(MsgChannel *c) const
     *c << flags;
 }
 
-LoginMsg::LoginMsg(unsigned int myport, const std::string &_nodename, const std::string _host_platform)
+LoginMsg::LoginMsg(unsigned int myport, const std::string &_nodename, const std::string &_netname, const std::string _host_platform)
```

Comment:
We shouldn't need that.  It isn't the schedulers problem if the client connects to the wrong one, it is a client problem.  

In fact this is a valid and legal case: if someone specifies a specific scheduler to attach to, then matching the netname is not required.

This is also a breaking change which makes this version incompatible with older clients.  I do not want to make such a change.